client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -338,4 +338,78 @@ class Commit(
     else
       createCommit(message = message, diffs = diff.blobDiffs, commitBase = id)
   }
+
+  /** Generates a stream of outputs in this commit by walking through its folder tree.
+   *  The stream ends at the first failure, or when there are no folders left. If the commit is not saved, its only element is that Failure
+   *  @param walker the FolderWalker to process the folders in the folder tree.
+   *  @return a stream of Try's of WalkOutput.
+   */
+  def walk[T](walker: FolderWalker[T])(implicit ec: ExecutionContext): Stream[Try[T]] = {
+    if (!saved)
+      Stream(Failure(new IllegalCommitSavedStateException("Commit must be saved before it can be walked")))
+    else
+      continueWalk(getFolder(PathList(Nil)).get, walker)
+  }
+
+  /** Get the folder corresponding to a path in list form
+   *  @param location location
+   *  @return the folder, if succeeds
+   */
+  def getFolder(location: PathList)(implicit ec: ExecutionContext): Try[Folder] = {
+    clientSet.versioningService.GetCommitComponent2(
+      commit_sha = id.get,
+      repository_id_repo_id = repo.id,
+      location = if (location.components.length > 0) Some(location.components) else None
+    ).flatMap(r => {
+        val folderPath = location.path
+        val responseFolder = r.folder
+
+        val folderNames = responseFolder
+        .flatMap(_.sub_folders) // Option[List[VersioningFolderElement]]
+        .map(_.map(folder => folder.element_name.get))
+        // Option[List[String]]
+        .map(_.sortWith((name1: String, name2: String) => name1 < name2))
```

Comment:
There should be a standard sorter, no? No big deal, I was just surprised.

Reply:
You're right actually, as String has an implicit ordering we can just use List.sorted. Fixed.